### ISSUE TITLE
Localized month gets capitalized

### ISSUE THREAD

**Issue by Dambakk** [korlibs/klock#146]
Hi! 

I'm defining a custom KlockLocale for Norwegian and setting the default KlockLocale to it's object: `KlockLocale.default = Norwegian()`

`Norwegian` has defined it's months in lowercase, just like the examples in the docs. However, when converting a date to string like this:
```
date.toString(DateFormat("d. MMMM yyyy"))
```
the returned string contains a capitalized month name (which is not correct for Norwegian, and several other languages).

When running `Month.January.localName` it returns the month lowercased. 

Is this an issue with Klock or anything I do wrong?

Otherwise, very happy with Klock! :)

**Comment by soywiz** [comment]
Maybe there is a capitalize somewhere. Can you make a PR with the Norwegian locale? If possible, making a failing test where it shows the right format

If there is a capitalize somewhere maybe we can either add a boolean flag indicating if that capitalize should be performed or create a separate list defaulting to a capitalized version of the names, that can be overriden to just the lowercased list

**Comment by Dambakk** [comment]
Output from failing test:
```
expected:<...9 21:36
Onsdag, 13. [mars 2019
13. m]ars 2019
13.03.2019
...> but was:<...9 21:36
Onsdag, 13. [Mars 2019
13. M]ars 2019
13.03.2019
...>
```

**Comment by Dambakk** [comment]
I think maybe the issue is bigger than I first anticipated. Full disclosure, I used Google Translate to verify the translations in the tests (so I/Google might be wrong) but several of them had lowercased months (e.g. spanish, dutch, portuguese). 

I found that PatternDateFormat.kt, lines 140-142 does some capitalization on months. I changed it to use localized months instead, as explained in the documentation, and fixed the failing tests while cross-checking with google. I think this approach will work if the months in the KlockLocale classes are written in the form that is correct for that locale (lowercased or capitalized).

Turns out, all this also applies to weekdays.

I've added another commit to the PR that changes the `format` method in `PatternDateFormat` to use localized names for days and months, and checked and updated the localized values for the failing tests, but someone should verify the changes.

Also, I am up for contributing more! Just let me know! :)
